Api/Controllers/ProductsController.cs

```diff
@@ -0,0 +1,194 @@
+                                  ILogger<ProductsController> logger,
+                                  LinkGenerator linkGenerator)
+        {
+            _categoryService = categoryService;
+            _productService = productService;
+            _logger = logger;
+            _linkGenerator = linkGenerator;
+        }
+
+        // GET: api/<ProductsController>
+        [HttpGet]
+        [Produces("application/json")]
+        [ProducesResponseType(typeof(LinkCollectionWrapper<ProductModel>), StatusCodes.Status200OK)]
+        public async Task<IActionResult> Get(int categoryId, int page = 0, int limit = 50)
+        {
+            var dTOs = await _productService.GetAllProductsByCategoryIdAsync(categoryId, page, limit);
+
+            var models = dTOs.Select(ProductModel.MapFrom).ToList();
+
+            foreach (var product in models)
+            {
+                var links = CreateLinksForProduct(categoryId, product.Id);
+                product.Links = links.ToList();
+            }
+
+
+            var productsWrapper = new LinkCollectionWrapper<ProductModel>(models);
+            return Ok(CreateLinksForGetAllProducts(productsWrapper));
+        }
+
+        // GET api/<ProductsController>/5
+        [HttpGet("{id}")]
+        [Produces("application/json")]
+        [ProducesResponseType(typeof(ProductModel), StatusCodes.Status200OK)]
+        [ProducesResponseType(StatusCodes.Status404NotFound)]
+        public async Task<IActionResult> GetById(int categoryId, int id)
+        {
+            var productDto = await _productService.GetProductByIdAsync(id);
+            var categoryDto = await _categoryService.GetCategoryByIdAsync(categoryId);
+
+            if (productDto == null || categoryDto == null)
+                return NotFound();
+
+            var model = ProductModel.MapFrom(productDto);
+            model.Links = CreateLinksForProduct(categoryId, model.Id).ToList();
+
+            return Ok(model);
+        }
+
+        // POST api/<ProductsController>
+        [HttpPost]
+        [Consumes("application/json")]
+        [Produces("application/json")]
+        [ProducesResponseType(typeof(ProductModel), StatusCodes.Status201Created)]
+        [ProducesResponseType(StatusCodes.Status404NotFound)]
+        [ProducesResponseType(StatusCodes.Status400BadRequest)]
+        public async Task<IActionResult> Post(int categoryId, [FromBody] ProductWriteModel model)
+        {
+            var categoryDto = await _categoryService.GetCategoryByIdAsync(categoryId);
+
+            if (categoryDto == null)
+                return NotFound();
+
+            var dto = ProductWriteModel.MapTo(model);
+            dto.CategoryId = categoryId;
+
+            ProductDTO? addedDto = null;
+
+            try
+            {
+                addedDto = await _productService.AddProductAsync(dto);
+            }
+            catch (Exception ex)
+            {
+                _logger.LogError(ex, $"Add product error");
```

Comment:
same as with other try / catch blocks like this one, I don't think it adds any value